avoid bundling unnecesary languages, except `graphql`/`json` (#3285)
graphql/graphiql: packages/monaco-graphql/README.md

- "linting" => "diagnostics" in lsp terminology
- the default keymap is different, more vscode like
- command palette and right click context menu are important

> - you can extend the standard completion/linting/etc. provided. for example,

Reviewer: ```suggestion
- you can extend the provided standard completion, linting, etc. for example,
```

  `editor.setModelMarkers()`
- [Monaco Editor API Docs](https://microsoft.github.io/monaco-editor/api/index.html)
- [Monaco Editor Samples](https://github.com/Microsoft/monaco-editor-samples)
  repository is great for tips on implementing with different bundlers,
  runtimes, etc.

## Avoid Bundle All `monaco-editor`'s Languages

While importing `monaco-editor` in your project, you silently import 83 builtin

> languages, such as `typescript`, `html`, `css`, `json` and other. You can found

Reviewer: ```suggestion
languages, such as `typescript`, `html`, `css`, `json` and others. You can found
```

a full list of
[basic-languages](https://github.com/microsoft/monaco-editor/tree/main/src/basic-languages)
and
[languages](https://github.com/microsoft/monaco-editor/tree/main/src/language).

For `monaco-graphql` basically you need only 2 languages - `graphql` and `json`.
To improve performance and drop loading unused languages starting from
monaco-graphql@1.3.0 version you can do it by replacing all your `monaco-editor`
exports with `monaco-graphql/esm/monaco-editor` that will load only `graphql`

> and `json` languages.

Reviewer: ```suggestion
For `monaco-graphql`, you need only 2 languages - `graphql` and `json`.
In version monaco-graphql@1.3.0 and later, you can replace all `monaco-editor`
exports with `monaco-graphql/esm/monaco-editor` to improve performance, load
only `graphql` and `json` languages, and skip loading unused languages.
```

Author: can commit, since changed little but, but applied in my commit, thx

```diff
-import { ... } from 'monaco-editor'
+import { ... } from 'monaco-graphql/esm/monaco-editor'
```

### Catch Future Import Mistakes with ESLint

> To prevent mis importing of `monaco-editor` you can setup default

Reviewer: ```suggestion
To prevent mis-importing of `monaco-editor`, you can set up default
```

`no-restricted-imports` rule for JavaScript projects or
`@typescript-eslint/no-restricted-imports` for TypeScript projects.

```json5
{
  rules: {
    // or @typescript-eslint/no-restricted-imports
    'no-restricted-imports': [
      'error',
      {
        name: 'monaco-editor',

>         message: '`monaco-editor` import all languages; use `monaco-graphql/esm/monaco-editor` instead to import only `json` and `graphql` languages',

Reviewer: ```suggestion
        message: '`monaco-editor` imports all languages; use `monaco-graphql/esm/monaco-editor` instead to import only `json` and `graphql` languages',
```
